File: Backend/Backend/DTOs/TaskListDto.cs
using Backend.Models;

namespace Backend.DTOs
{
    public class TaskListDto
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime DeletedOn { get; set; }

        //References
        //public string CreatorId { get; set; }
        public virtual AppUserDto Creator { get; set; }
        public virtual ICollection<TaskItemDto> Tasks { get; set; }
        public virtual ICollection<AppUserDto> AssignedMembers { get; set; }
    }
}

File: Frontend/Frontend/src/app/_dtos/taskListDto.ts
import { AppUserDto } from "./appUserDto"
import { TaskItemDto } from "./taskItemDto"

export class TaskListDto{
    public id: string = ''
    public name: string = ''
    public description: string = ''
    //public creatorId: string = ''
    public creator: AppUserDto = new AppUserDto()
    public tasks: Array<TaskItemDto> = []
    public assignedMembers: Array<AppUserDto> = []
}

/*
"id": "string",
  "name": "string",
  "description": "string",
  "isDeleted": true,
  "deletedOn": "2024-03-14T11:52:18.324Z",
  "creatorId": "string",
  "creator": {
    "id": "string",
    "userName": "string",
    "image": "string",
    "firstName": "string",
    "lastName": "string"
  },
*/